esteid/middleware.py

```diff
@@ -1,187 +0,0 @@
-import logging
-import warnings
-
-from django.utils.cache import patch_vary_headers
-
-from esteid.helpers import parse_legacy_dn, parse_rfc_dn
-from . import config
-from .ocsp import OCSPVerifier
-
-
-try:
-    from django.utils.deprecation import MiddlewareMixin
-except ImportError:  # Django < 1.10
-    # Works perfectly for everyone using MIDDLEWARE_CLASSES
-    MiddlewareMixin = object
-
-
-logger = logging.getLogger(__name__)
-
-
-ERROR_NO_DN = 3000
-ERROR_INVALID_DN = 3001
-
-
-class MultiHostMiddleware(MiddlewareMixin):
-    def process_request(self, request):
-        # Choose which subhost to use (if any):
-        host = request.META["HTTP_HOST"]
-        if host[-3:] == ":80":
-            host = host[:-3]  # ignore default port number, if present
-
-        request.subhost_name = 'default'
-        for host_name, host_config in config.get_hosts().items():
-            if host.startswith(host_name) or host == host_config.get('hostname'):
-                request.subhost_name = host_name
-                request.urlconf = host_config.get('urlconf')
-
-    def process_response(self, request, response):
-        if getattr(request, "urlconf", None):
-            patch_vary_headers(response, ('Host',))
-
-        return response
-
-
-class BaseIdCardMiddleware(MiddlewareMixin):
```

Comment:
Was there any particular reason you removed this?

Reply:
Hmm. Probably I didn't look at it carefully enough. I guess I could reinstate it, but is there any documentation how to use it, so it wouldn't be here useless? Like how to set up nginx/apache.

Reply:
The documentation for nginx and apache lives in the esteid documentation provided by sk.ee and is out of scope for this library (see links here https://eid.eesti.ee/index.php/Kasutaja_tuvastamine_veebis#Veebiserveri_konfigureerimine).

This middleware exists to help set up the django side quickly and I believe it makes sense to keep it for now.

From usage side I know for a fact this is used by ecoop.

Reply:
Thanks, this is exactly the answer I was looking for. Does this middleware remain useful in the future? As I understand, probably it was a temporary solution, and messing with web server configuration for the sake of supporting only ID card isn't something one would be doing these days?